ts/screens/wallet/payment/ScanQrCodeScreen.tsx

```diff
@@ -247,9 +247,12 @@ class ScanQrCodeScreen extends React.Component<Props, State> {
         O.chain(assets => AR.head([...assets]))
       );
       if (O.isSome(maybePickedImage)) {
-        ReaderQR.readerQR(maybePickedImage.value.uri)
-          .then((data: string) => {
-            this.onQrCodeData(data);
+        RNQRGenerator.detect({ uri: maybePickedImage.value.uri })
+          .then(data => {
+            if (data.values.length === 0) {
+              this.onInvalidQrCode();
+            }
+            this.onQrCodeData(data.values[0]);
```

Comment:
`onQrCodeData` either returns `void` in case of a success or `() => void` in case of a failure (_e.g., on valid QRCode with unrelated content).

However, since latter case returns a function, the same is never executed and the UI does not shows the appropriate feedback